ui/components/app/snaps/snap-ui-renderer/components/box.ts

```diff
@@ -49,5 +50,6 @@ export const box: UIComponentFactory<BoxElement> = ({
     alignItems: element.props.center && AlignItems.center,
     className: 'snap-ui-renderer__panel',
     color: TextColor.textDefault,
+    flexWrap: FlexWrap.Wrap, // ensures components can wrap horizontally
```

Comment:
Both this change and the `flex` change will have impact beyond the overflow we are trying to fix here.

Have you tested existing Snap UI's to make sure this doesn't cause regressions in behavior?